test/pipboy/parser/composite_readers_test.rb

```diff
@@ -7,14 +8,25 @@ def setup
   end
 
   def test_header
+    headerBytes = StringIO.new("\xA0\x00\x00\x00\x03")
+    assert_equal([160,3], Pipboy::Parser::CompositeReaders.header(headerBytes))
   end
 
   def test_list
+    primitiveReaders = Pipboy::Parser::PrimitiveReaders
+    streamBytes = StringIO.new("\x03\x00\x01\x00\x00\x00\x02\x00\x00\x00\x03\x00\x00\x00")
+    assert_equal([1, 2, 3], Pipboy::Parser::CompositeReaders.list(streamBytes, primitiveReaders, :uint32))
   end
 
   def test_dict_entry
+    streamBytes = StringIO.new("\xA0\x00\x00\x00\x03\x48\x45\x4C\x4C\x4F\x57\x4F\x52\x4C\x44\x70\x69\x70\x62\x6f\x79\x00\x00\x00\x00\x00")
+    assert_equal({"\u0003HELLOWORLDpipboy" => 160}, Pipboy::Parser::CompositeReaders.dict_entry(streamBytes))
```

Comment:
This is on the right track! That 0x03 in the 5th byte (where the string representing the key should begin) seems extraneous. Also, the string reader (PrimitiveReaders.string) will read until the first 0x0, so you only need one `\x00` at the end of the byte stream (right now there are 5).

You might also pick a shorter string to keep the byte stream shorter, e.g.

```
 assert_equal({"pip" => 160}, Pipboy::Parser::CompositeReaders.dict_entry(streamBytes))
```